packages/smart-wallet/src/walletFactory.js

```diff
@@ -122,19 +123,68 @@ export const start = async (zcf, privateArgs) => {
     E(walletBridgeManager).setHandler(handleWalletAction));
 
   // Resolve these first because the wallet maker must be synchronous
-  const getInvitationIssuer = E(zoe).getInvitationIssuer();
+  const invitationIssuerP = E(zoe).getInvitationIssuer();
   const [invitationIssuer, invitationBrand, publicMarshaller] =
     await Promise.all([
-      getInvitationIssuer,
-      E(getInvitationIssuer).getBrand(),
+      invitationIssuerP,
+      E(invitationIssuerP).getBrand(),
       E(board).getReadonlyMarshaller(),
     ]);
 
+  const makeAssetRegistry = () => {
```

Comment:
this has access to all of `start` scope, making it a little harder to read and audit.

please move this function def to module scope or another file. consider unit testing it.

Reply:
Hosting it is a good idea... done.

As to unit testing... I'm inclined to let the existing tests suffice. Mocking an `anyBank` would be tedious, and using the existing vbank code would be more or less another integration test like the others.